ui/Screen/Project/Source.svelte

```diff
@@ -1,148 +1,132 @@
-<script>
-  import { getContext } from "svelte";
-  import { format } from "timeago.js";
-
+<script lang="typescript">
+  import Router, { push } from "svelte-spa-router";
+
+  import { openPath } from "../../src/ipc";
+  import type { HorizontalItem } from "../../src/menu";
+  import * as notification from "../../src/notification";
+  import * as path from "../../src/path";
+  import { checkout } from "../../src/project";
+  import type { Project, User } from "../../src/project";
   import {
-    currentPeerId,
-    currentRevision,
-    object as objectStore,
-    ObjectType,
-    objectPath,
-    objectType,
-    readme,
-    resetObjectPath,
-    resetObjectType,
-    fetchObject,
-  } from "../../src/source.ts";
-
-  import { EmptyState, Remote } from "../../DesignSystem/Component";
-
-  import FileSource from "../../DesignSystem/Component/SourceBrowser/FileSource.svelte";
-  import CommitTeaser from "../../DesignSystem/Component/SourceBrowser/CommitTeaser.svelte";
-  import Readme from "../../DesignSystem/Component/SourceBrowser/Readme.svelte";
-  import Folder from "../../DesignSystem/Component/SourceBrowser/Folder.svelte";
-
-  const project = getContext("project");
-
-  const reset = () => {
-    resetObjectPath();
-    resetObjectType();
+    fetch,
+    selectRevision,
+    store,
+  } from "../../src/screen/project/source";
+  import type { Revision } from "../../src/source";
+  import * as screen from "../../src/screen";
+
+  import IconCommit from "../../DesignSystem/Primitive/Icon/Commit.svelte";
+  import IconHouse from "../../DesignSystem/Primitive/Icon/House.svelte";
+
+  import ActionBar from "../../DesignSystem/Component/ActionBar.svelte";
+  import HorizontalMenu from "../../DesignSystem/Component/HorizontalMenu.svelte";
+  import Remote from "../../DesignSystem/Component/Remote.svelte";
+  import RevisionSelector from "../../DesignSystem/Component/SourceBrowser/RevisionSelector.svelte";
```

Comment:
See above. I find it easier to read and edit if they are grouped. What is your reason for importing them one by one?

Reply:
I disagree, I find that extremely unproductive and necessitates maintaining indexs files, which require manual book-keeping.

Reply:
> @xla: IMHO it just shifts around where you do the bookkeeping.

It does not tho, when adding a component the implementer has to remember to add extra **entries** in the respective index file(s). None of that overhead is present when using the component directly.